### PR TITLE
fix: copyright template should use current year

### PR DESCRIPTION
**Pull request by ryanbrainard**
### What does this PR do?

Currently, when running eslint in fix mode, it adds copyright headers with the date `2020` because that value is hardcoded in the template. This changes it to dynamically use the current year. This required changing the eslint config from JSON to JS.

### What issues does this PR fix or reference?
@W-13131914@

Fixes https://github.com/forcedotcom/salesforcedx-vscode/issues/4849

### Functionality Before
2020 copyright

### Functionality After
Current year copyright (only affects files currently without a copyright)

### CODE REVIEW

**Comment by ryanbrainard** [comment]
Since it's hard to see the diff with the file name and format change:

```diff
Index: Clipboard.txt
===================================================================
diff --git a/.eslint.json b/.eslintrc.js
rename from .eslint.json
rename to .eslintrc.js
--- a/.eslint.json	
+++ b/.eslintrc.js	
@@ -1,64 +1,73 @@
-{
-  "root": true,
-  "parser": "@typescript-eslint/parser",
-  "parserOptions": {
-    "ecmaVersion": 6,
-    "sourceType": "module"
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+const year = new Date().getFullYear();
+
+module.exports = {
+  root: true,
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module'
   },
-  "plugins": ["@typescript-eslint", "jsdoc", "eslint-plugin-header"],
-  "extends": [
-    "plugin:@typescript-eslint/recommended",
-    "plugin:prettier/recommended"
+  plugins: ['@typescript-eslint', 'jsdoc', 'eslint-plugin-header'],
+  extends: [
+    'plugin:@typescript-eslint/recommended',
+    'plugin:prettier/recommended'
   ],
-  "rules": {
-    "ban-ts-ignore": "off",
-    "@typescript-eslint/ban-ts-ignore": "off",
-    "@typescript-eslint/no-unused-vars": "off",
-    "camelcase": "off",
-    "@typescript-eslint/camelcase": "off",
-    "constructor-super": "warn",
-    "curly": "error",
-    "eqeqeq": "error",
-    "no-buffer-constructor": "error",
-    "no-caller": "error",
-    "no-debugger": "warn",
-    "no-duplicate-case": "error",
-    "no-duplicate-imports": "error",
-    "no-eval": "error",
-    "no-extra-semi": "warn",
-    "no-redeclare": "error",
-    "no-sparse-arrays": "error",
-    "no-throw-literal": "error",
-    "no-unsafe-finally": "warn",
-    "no-unused-labels": "warn",
-    "no-restricted-globals": [
-      "warn",
-      "name",
-      "length",
-      "event",
-      "closed",
-      "external",
-      "status",
-      "origin",
-      "context"
+  rules: {
+    'ban-ts-ignore': 'off',
+    '@typescript-eslint/ban-ts-ignore': 'off',
+    '@typescript-eslint/no-unused-vars': 'off',
+    camelcase: 'off',
+    '@typescript-eslint/camelcase': 'off',
+    'constructor-super': 'warn',
+    curly: 'error',
+    eqeqeq: 'error',
+    'no-buffer-constructor': 'error',
+    'no-caller': 'error',
+    'no-debugger': 'warn',
+    'no-duplicate-case': 'error',
+    'no-duplicate-imports': 'error',
+    'no-eval': 'error',
+    'no-extra-semi': 'warn',
+    'no-redeclare': 'error',
+    'no-sparse-arrays': 'error',
+    'no-throw-literal': 'error',
+    'no-unsafe-finally': 'warn',
+    'no-unused-labels': 'warn',
+    'no-restricted-globals': [
+      'warn',
+      'name',
+      'length',
+      'event',
+      'closed',
+      'external',
+      'status',
+      'origin',
+      'context'
     ], // non-complete list of globals that are easy to access unintentionally
-    "no-var": "error",
-    "jsdoc/no-types": "warn",
-    "@typescript-eslint/semi": "warn",
-    "header/header": [
+    'no-var': 'error',
+    'jsdoc/no-types': 'warn',
+    '@typescript-eslint/semi': 'warn',
+    'header/header': [
       2,
-      "block",
+      'block',
       [
-        "",
+        '',
         {
-          "pattern": " \\* Copyright \\(c\\) \\d{4}, salesforce\\.com, inc\\.",
-          "template": " * Copyright (c) 2020, salesforce.com, inc."
+          pattern: ' \\* Copyright \\(c\\) \\d{4}, salesforce\\.com, inc\\.',
+          template: ` * Copyright (c) ${year}, salesforce.com, inc.`
         },
-        " * All rights reserved.",
-        " * Licensed under the BSD 3-Clause license.",
-        " * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause",
-        " "
+        ' * All rights reserved.',
+        ' * Licensed under the BSD 3-Clause license.',
+        ' * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause',
+        ' '
       ]
     ]
   }
-}
+};
```

File: packages/salesforcedx-vscode-soql/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    'jsdoc/no-types': 'warn',
+    '@typescript-eslint/semi': 'warn',
+    'header/header': [
+      2,
+      'block',
+      [
+        '',
+        {
+          pattern: ' \\* Copyright \\(c\\) \\d{4}, salesforce\\.com, inc\\.',
+          template: ` * Copyright (c) ${year}, salesforce.com, inc.`
```
Comment (ryanbrainard) [inline packages/salesforcedx-vscode-soql/.eslintrc.js:64]: This line is the actual change with `${year}` being passed in.

**Review by jonnyhork — APPROVED** [review]
Awesome, cool to see the use of a JS file instead of JSON

**Comment by randi274** [comment]
@ryanbrainard taking a review pass on this today - am I missing a step to verify?

- Make a new file in `packages/salesforcedx-vscode-soql` or remove header from an existing file
- `npm run lint:fix` at the top level or inside soql package
- Nothing happens?

I'd expect the header to at least get added for me (even with the wrong year) which I think it would've been doing before, so I think I'm missing something.

**Comment by ryanbrainard** [comment]
> @ryanbrainard taking a review pass on this today - am I missing a step to verify?
> 
> * Make a new file in `packages/salesforcedx-vscode-soql` or remove header from an existing file
> * `npm run lint:fix` at the top level or inside soql package
> * Nothing happens?
> 
> I'd expect the header to at least get added for me (even with the wrong year) which I think it would've been doing before, so I think I'm missing something.

@randi274 Thanks for taking a look. Make sure you the file you create is a `.ts` file in `src` and then it should work. The rule applies everywhere in the package when in the editor, but it is only enforced and fixed for `src/**/*.ts` as the `--ext .ts ./src` flags are set.

**Review by randi274 — APPROVED** [review]

**Pull request merged**